File: app/src/main/java/ru/SnowVolf/convertx/ui/fragments/extended/hash/MdSha.java
package ru.SnowVolf.convertx.ui.fragments.extended.hash;

import android.os.Bundle;
import android.support.v7.view.menu.MenuBuilder;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import ru.SnowVolf.convertx.R;
import ru.SnowVolf.convertx.algorhitms.ChecksumAlgs;
import ru.SnowVolf.convertx.settings.Preferences;
import ru.SnowVolf.convertx.ui.Toasty;
import ru.SnowVolf.convertx.ui.fragments.base.BaseFragment;

/**
 * Created by Snow Volf on 14.02.2017.
 */

public class MdSha extends BaseFragment {
    public EditText data;
    public TextView dataOut;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState){
        rootView = inflater.inflate(R.layout.fragment_ex, container, false);
        data = (EditText) rootView.findViewById(R.id.exData);
        data.setHint(R.string.hint_md_sha);
        data.setFocusable(true);
        dataOut = (TextView) rootView.findViewById(R.id.exText);
        return rootView;
    }

    @Override
    public void onCreate(Bundle savedInstanceState){
        super.onCreate(savedInstanceState);

        setHasOptionsMenu(true);
    }

    public void onActivityCreated(Bundle savedInstanceState){
        super.onActivityCreated(savedInstanceState);
        TITLE = R.string.md_sha;
        data.setTypeface(Mono);
        data.setTextSize(Preferences.getFontSize());
        dataOut.setTypeface(Mono);
        data.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {
            }

            @Override
            public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
            }

            @Override
            public void afterTextChanged(Editable editable) {
                String source = data.getText().toString();
                /**
                 * Алгоритмы в ChecksumAlgs, чтоб код не заграждать
                 */
                dataOut.setText(new StringBuilder()
                        //фикс предупреждения Lint: [do not concentrate text displayed with setText. Use resources string with placeholders] в Android Studio
                        .append("MD5:\n").append(ChecksumAlgs.INSTANCE.MD5(source))
                        + "\n\nSHA-1:\n" + ChecksumAlgs.INSTANCE.SHA1(source)
                        + "\n\nSHA-224:\n" + ChecksumAlgs.INSTANCE.SHA224(source)
                        + "\n\nSHA-384:\n" + ChecksumAlgs.INSTANCE.SHA384(source)
                        + "\n\nSHA-512:\n" + ChecksumAlgs.INSTANCE.SHA512(source));
            }
        });
    }

    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
        if (menu != null) {
            menu.clear();
        } else
            menu = new MenuBuilder(getContext());
        //добавляем пункты меню
        menu.add(R.string.clear_all)
                .setIcon(R.drawable.ic_menu_clear_all)
                .setShowAsActionFlags(MenuItem.SHOW_AS_ACTION_ALWAYS)
                .setOnMenuItemClickListener(menuItem -> {
                    clearAllText();
                    return true;
                });
    }
    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home)
            getActivity().finish();
        return true;
    }
    public void clearAllText(){
        data.setText("");
        dataOut.setText("");
        Toasty.success(getContext(), getString(R.string.cleared), Toast.LENGTH_SHORT, true).show();
    }
}

File: app/src/main/java/ru/svolf/convertx/ui/fragments/extended/hash/MdSha.kt
package ru.svolf.convertx.ui.fragments.extended.hash


import ru.svolf.convertx.algorhitms.Checksum.MD5
import ru.svolf.convertx.algorhitms.Checksum.SHA1
import ru.svolf.convertx.algorhitms.Checksum.SHA224
import ru.svolf.convertx.algorhitms.Checksum.SHA384
import ru.svolf.convertx.algorhitms.Checksum.SHA512
import ru.svolf.convertx.ui.fragments.base.BaseFragment
import android.widget.EditText
import android.widget.TextView
import android.os.Bundle
import ru.svolf.convertx.R
import android.text.TextWatcher
import android.text.Editable
import android.view.*
import ru.svolf.convertx.ui.Toasty
import android.widget.Toast
import ru.svolf.convertx.settings.Preferences
import java.lang.StringBuilder

/**
 * Created by Snow Volf on 14.02.2017.
 */
class MdSha : BaseFragment() {
    var data: EditText? = null
    var dataOut: TextView? = null
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        rootView = inflater.inflate(R.layout.fragment_ex, container, false)
        data = rootView.findViewById<View>(R.id.exData) as EditText
        data!!.setHint(R.string.hint_md_sha)
        data!!.isFocusable = true
        dataOut = rootView.findViewById<View>(R.id.exText) as TextView
        return rootView
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setHasOptionsMenu(true)
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        activityTitle = R.string.md_sha

        data!!.textSize = Preferences.fontSize.toFloat()

        data!!.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {}
            override fun onTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {}
            override fun afterTextChanged(editable: Editable) {
                val source = data!!.text.toString()
                /**
                 * Алгоритмы в ChecksumAlgs, чтоб код не заграждать
                 */
                dataOut!!.text =
                    (StringBuilder() //фикс предупреждения Lint: [do not concentrate text displayed with setText. Use resources string with placeholders] в Android Studio
                        .append("MD5:\n").append(MD5(source))
                        .toString() + "\n\nSHA-1:\n" + SHA1(source)
                            + "\n\nSHA-224:\n" + SHA224(source)
                            + "\n\nSHA-384:\n" + SHA384(source)
                            + "\n\nSHA-512:\n" + SHA512(source))
            }
        })
    }

    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
        menu.clear()
        //добавляем пункты меню
        menu.add(R.string.clear_all)
            .setIcon(R.drawable.ic_menu_clear_all)
            .setShowAsActionFlags(MenuItem.SHOW_AS_ACTION_ALWAYS)
            .setOnMenuItemClickListener { menuItem: MenuItem? ->
                clearAllText()
                true
            }
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) activity!!.finish()
        return true
    }

    fun clearAllText() {
        data!!.setText("")
        dataOut!!.text = ""
        Toasty.success(context!!, getString(R.string.cleared), Toast.LENGTH_SHORT, true).show()
    }
}